### PR TITLE
Adding openstack_shares_snapshots metrics

### PR DESCRIPTION
**Pull request by rajivmucheli**
@Carthaca @fwiesel 

this is wrt https://issues.wdf.sap.corp/browse/CCM-1144

Do let me know if its fine ?

### CODE REVIEW

**Comment by Carthaca** [comment]
Hi, I think grouping by share_id is not necessary.
But it would be good to also extract the snapshot status out of the `share_snapshot_instances` table.
And please adjust the descriptions to match that this is for snapshots.

File: manila/values.yaml [hunk 1]
```diff
@@ -80,7 +80,20 @@ postgres:
...
+        metrics:
+          - snapshot_id:
+              usage: "LABEL"
+              description: "SnapShot ID"
+          - status:
+              usage: "LABEL"
+              description: "Snapshot Status"
+          - count_gauge:
+              usage: "GAUGE"
+              description: "Number of share networks"
```
Comment (Carthaca) [inline manila/values.yaml]: typo -> `Number of share snapshots`

File: manila/values.yaml [hunk 2]
```diff
@@ -80,7 +80,20 @@ postgres:
           - size_gauge:
               usage: "GAUGE"
               description: "Size of shares"
+       openstack_shares_snapshots:
+         query: "SELECT COUNT(snapshot_id) AS count_gauge, status FROM share_snapshot_instances GROUP BY status"
+        metrics:
+          - snapshot_id:
```
Comment (fwiesel) [inline manila/values.yaml]: The snapshot_id is not part of the query output, so it's not a label.

**Review by fwiesel — APPROVED** [review]
LGTM

**Comment by abattye** [comment]
Please can we stop adding these metrics that cause restarts to ther DB pods when changes are made to the metric query. Its got to be seen as an anti-pattern. Instead we need to devise an approach where a seperate metric pod is created to run these queries, independent of the core DB pod which in general we shoud view as immutable.

**Comment by rajivmucheli** [comment]
@abattye 

Maurice & I, have discussed this approach last week, all services using postgres-db (via postgres-exporter) have metrics container in their pods, while mariadb services (eg. designate) dont have metrics container. We are currently working on this plan. It will be helpful if you could suggest further on how we can expedite this implementation!

**Comment by BugRoger** [comment]
The metrics collection must be decoupled from the database. 

See: https://issues.wdf.sap.corp/browse/CCM-1153

**Comment by BugRoger** [comment]
Please don't add or change further metrics until the above task is implemented.

**Pull request merged**